client/src/components/DateInput.js

```diff
@@ -31,19 +26,17 @@ const DateInput = ({
         {label + (required ? " *" : "")}
         <input
           type="checkbox"
-          checked={autoCompleteCheckbox}
-          onClick={handleClick}
+          checked={showAutocomplete}
+          onClick={setShowAutocomplete.bind(null, undefined)}
```

Comment:
This can also be `() => setShowAutocomplete` (arrow functions don't rebind `this`)